src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,46 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    private final BuldozerProducer buldozerProducer = new BuldozerProducer();
+    private final ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+    private final TruckProducer truckProducer = new TruckProducer();
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            List<? extends Machine> machines = buldozerProducer.get();
+            return new ArrayList<>(machines);
+        } else if (type == Excavator.class) {
+            List<? extends Machine> machines = excavatorProducer.get();
+            return new ArrayList<>(machines);
+        } else if (type == Truck.class) {
+            List<? extends Machine> machines = truckProducer.get();
+            return new ArrayList<>(machines);
```

Comment:
i've mentioned in previous comment also that these variables is redundant
```suggestion
            return new ArrayList<>(buldozerProducer.get());
        } else if (type == Excavator.class) {
            return new ArrayList<>(excavatorProducer.get());
        } else if (type == Truck.class) {
            return new ArrayList<>(truckProducer.get());
```